_posts/2024-01-08-confidence-calibration.md

```diff
@@ -0,0 +1,118 @@
+
+Temperature scaling involves tuning a softmax temperature value to minimize Negative Log-likelihood loss on a held-out validation set. This value is then used to *soften* the output of the softmax layer. 
+
+$$
+\begin{equation}
+\text{Softmax}(y) = \frac{\exp(y_i)}{\sum_j \exp(y_j)}
+\end{equation}
+$$
+
+$$
+\begin{equation}
+\text{Temp-Softmax}(y) = \frac{\exp(y_i/T)}{\sum_j \exp(y_j/T)}
+\end{equation}
+$$
+
+The intuition behind temperature scaling is that the `T` value penalizes high probability scores thereby resulting in better confidence in these values.
+
+After tuning a temperature scaling value on our validation dataset, we noticed that our model was better calibrated(Lower ECE and MCE values as well) and was able to better classify low-confidence score predictions.
+
+| ![image info](../assets/images/confidence-calibration-blog/cal-score-plot.png) | 
+|:--:| 
+| *Post calibration, we notice that we are able to better classify low-confidence scores under wrong prediction bins.* |
+
+
+| ![image info](../assets/images/confidence-calibration-blog/old_reliability_graph.png) | 
+|:--:| 
+| *Old Reliability Graph on Validation Set* |
+
+| ![image info](../assets/images/confidence-calibration-blog/new_reliability_graph.png) | 
+|:--:| 
+| *New Reliability Graph on Validation Set* |
+
+
+## Thresholding
+
+Keeping our current objective in mind where we reject low-confidence predictions, we realized that thresholding individual intents on a temperature-scaled model could work quite well for us. We wanted an increment in our precision numbers without hitting recall for our intents. This is because we want to maximize our confidence in the current prediction without inadvertently increasing our **False Negatives** i.e. we still want to be accurate while predicting our positive class (could be any intent here). We plotted precision-recall curves with thresholds on the x-axis and precision/recall on the y-axis. We have default recipes to generate these threshold values that ensure that we maximize precision without affecting the recall. The data scientist or an ML engineer then can have a look at these plots and accordingly decide which threshold values to go with if they feel that 
+
+| ![image info](../assets/images/confidence-calibration-blog/precision-recall-curve.png) | 
+|:--:| 
+| *A precision-recall curve for the `_confirm_` intent.* |
+
+# Results
+
+After performing the above experiments on our datasets, we noticed a bump of ********************10 points******************** in macro-precision without affecting the macro-recall (+-loss of 1 point in recall). Our model became more robust to misclassifications and was able to reject low-confidence predictions thereby increasing the confidence in the model to handle sensitive compliance-related turns. Our product metrics also improved by a margin. After implementing the above solution we were able to bring down false consumer verification from **~0.67% to ~0.18%** of connected calls. 
```

Comment:
Can we use 'cases' instead of connected calls? It confuses the idea of whether this is production or a dataset. Same in the beginning of this post.

Reply:
changed references to point to a curated and biased dataset